src/smo/data/tuplet.ts

```diff
@@ -144,6 +144,30 @@ export class SmoTupletTree {
     }
   }
 
+  /**
+   * Determines whether two notes are part of the same tuplet.
+   *
+   * - Returns `true` if neither `noteOne` nor `noteTwo` is part of a tuplet.
+   * - Returns `true` if both notes are part of a tuplet and share the same `tupletId`.
+   * - Returns `false` otherwise.
+   *
+   * @param noteOne
+   * @param noteTwo
+   */
+  static areNotesPartOfTheSameTuplet(noteOne: SmoNote, noteTwo: SmoNote): boolean {
+    if (typeof(noteOne.tupletId) === 'undefined' && typeof(noteTwo.tupletId) === 'undefined') {
+      return true;
+    }
+    if (noteOne.tupletId === null && noteTwo.tupletId === null) {
+      return true;
+    }
+    if (noteOne.isTuplet && noteTwo.isTuplet && noteOne.tupletId == noteTwo.tupletId) {
+      return true;
+    }
+
```

Comment:
This is a little counter-intuitive.   They aren't really part of the a tuplet tree if null, are they?  Maybe separate the check for null and equality, or else rename the function to something like (areBothNullOrPartOfTheSameTupletTree).  

Also, what if one is undefined and the other is null?  Let's try to standardize on null/undefined checks using coalesce.  I know this is inconsistent elsewhere in the code, since the code predates coalesce, but something like 
``` typescript 
const areTupletsBothNull = (noteOne, noteTwo) => ((noteOne.tupletId ?? null) === null && ((noteTwo.tupletId ?? null) === null)
```
.  Maybe create 2 different checks, and if neither tuplet exists you don't need to go down

Reply:
Other than that, this looks good.  This is a big improvement for the tuplet beaming!

Should we open a different issue for the case of beam breaks in non-tuplets?  I'm thinking of something like 8 32nd notes, you might want a 16th beam after the first 4 notes.  I'm not sure what the correct default behavior should be, this might be something we let the user opt-in.

Reply:
Thanks for reviewing it. You made a couple of good points there, so yes, I will then further refactor this function.
About non tuplet beam breaks, I think it might be better to open another issue. And over there we can make a strategy on how to further improve beaming in general...